### PR TITLE
Apply misc small cleanups to unified scheduler

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

because I have some diff aesthetics, i need some preparation for #3946 

#### Summary of Changes

See individual hunks for explanations

extracted from #3946, see the pr for the general overview.

### CODE REVIEW

File: runtime/src/bank_forks.rs [hunk 1]
```diff
@@ -284,8 +284,8 @@ impl BankForks {
         self[self.highest_slot()].clone()
     }
 
-    pub fn working_bank_with_scheduler(&self) -> &BankWithScheduler {
-        &self.banks[&self.highest_slot()]
+    pub fn working_bank_with_scheduler(&self) -> BankWithScheduler {
+        self.banks[&self.highest_slot()].clone_with_scheduler()
```
Comment (ryoqun) [inline runtime/src/bank_forks.rs:288]: it turned out returning all call sites ended up calling .clone_with_scheduler() immediately after this. so, just do it inside here.

also, this makes it consistent the cousin (`BankForks::working_bank`), which isn't returning refs as well.

File: unified-scheduler-pool/src/lib.rs [hunk 2]
```diff
@@ -782,7 +781,7 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
             session_result_receiver,
             session_result_with_timings: None,
             scheduler_thread: None,
-            handler_threads: Vec::with_capacity(handler_count),
+            handler_threads: vec![],
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:784]: `handler_threads` is reassigned in `start_threads`, so no need to reserve capacity. also, this isn't perf sensitive code.

File: unified-scheduler-pool/src/lib.rs [hunk 3]
```diff
@@ -1152,54 +1151,56 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
...
-                let (task, sender) = select_biased! {
-                    recv(runnable_task_receiver.for_select()) -> message => {
-                        let Ok(message) = message else {
-                            break;
-                        };
-                        if let Some(task) = runnable_task_receiver.after_select(message) {
-                            (task, &finished_blocked_task_sender)
-                        } else {
-                            continue;
+            move || {
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:1154]: just indent stuff

File: unified-scheduler-pool/src/lib.rs [hunk 4]
```diff
@@ -551,7 +551,7 @@ mod chained_channel {
 
         pub(super) fn send_chained_channel(
             &mut self,
-            context: C,
+            context: &C,
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:554]: taking ref here will soon be desired in upcoming pr...

File: unified-scheduler-pool/src/lib.rs [hunk 5]
```diff
@@ -1441,7 +1442,7 @@ impl<TH: TaskHandler> InstalledScheduler for PooledScheduler<TH> {
 
 impl<S, TH> UninstalledScheduler for PooledSchedulerInner<S, TH>
 where
-    S: SpawnableScheduler<TH, Inner = PooledSchedulerInner<S, TH>>,
+    S: SpawnableScheduler<TH, Inner = Self>,
```
Comment (ryoqun) [inline unified-scheduler-pool/src/lib.rs:1445]: see 2 line above... ;)

File: unified-scheduler-pool/src/lib.rs (hunk 4 above)
Comment (apfitzge) [reply unified-scheduler-pool/src/lib.rs:554]: difference here is only that `context` is not dropped at the end of this function.

**Review by apfitzge — APPROVED** [review]
lgtm

**Pull request merged**